components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -1363,23 +1364,32 @@ public DataSection isSolvedBody(Patient patient)
             return null;
         }
         DataSection bodySection = new DataSection();
-        PatientData<String> patientData = patient.getData("solved");
+        PatientData<SolvedData> patientData = patient.getData("solved");
+        SolvedData data = patientData != null ? patientData.getValue() : null;
 
         int x = 0;
-        if (present.contains("solved")) {
-            String solved = patientData != null ? patientData.get("solved") : null;
+        if (present.contains(SolvedData.STATUS_PROPERTY_NAME)) {
+            String solved = data != null ? data.getStatus() : null;
```

Comment:
We're on Java 8, so now you can do `String solved = data?.getStatus();`

Reply:
It doesn't compile..

Reply:
Sorry, I meant to also give the link to the `Optional` documentation but forgot. Here's an article: http://www.oracle.com/technetwork/articles/java/java8-optional-2175753.html

Reply:
Still don't see how to use `data?.getStatus()` syntax with Optional or anything that would be as easy as `?.`, it seems to be from another language. In general I personally find explicit null checks more readable.

Reply:
`?.` syntax would be ideal in this case, but it seems to be a Groovy (?) and not Java construct? At the same time IMHO in simple cases like this `Optional` does not add much clarity or safety and is actually longer/harder to comprehend. IMHO it makes more sense for deep embedded values like in the example from the link above and/or when the optional value in question is used much further or much more in the code (so all those other places can benefit from the optional nature of the value or are forced to treat it as optional, even if by that line of code it is no longer obvious)